tests/integration/acp/nac/setup_then_start/collection_patch_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package test_acp_nac_setup_then_start
+
+import (
+	"testing"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/tests/action"
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+	"github.com/sourcenetwork/defradb/tests/state"
+)
+
+func TestNAC_GatesCollectionPatchPreSetup_AllowIfAuthorizedElseError(t *testing.T) {
+	test := testUtils.TestCase{
+		SupportedDatabaseTypes: immutable.Some(
+			[]state.DatabaseType{
+				// This test only supports file type databases since the setup steps will be done before
+				// the node is re-started with nac enabled (if it's in-memory it will loose setup state).
+				testUtils.BadgerFileType,
+			},
+		),
+		Actions: []any{
+			// Note: Since this is not an in-memory test, we can do the setup steps before nac is enabled.
```

Comment:
question: Why complicate things though?

Reply:
The comment exists in case people are confused why the setup is being done before here.

I was testing this because wanting to make sure that a feature like NAC on defradb restart, still works as expected and no bugs are slipped through that may mess with the `NAC` system to fail on restart.

IMO this is important to have as most tests just test single running db instance and not nac working properly even after restart, when setup was done before the restart.

Nonetheless, this seems to be too controversial, and in the hoping of not delaying in merging this PR furthur, I will accept the suggestion asking to remove this (for this and all other gates).

Reply:
- [ ] Remove this test